packages/e2e-tests/specs/inserter.test.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { sortBy, uniq } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	createNewPost,
+	insertBlock,
+	openGlobalInserter,
+} from '@wordpress/e2e-test-utils';
+
+describe( 'Inserter', () => {
+	const openAllInserterCategories = async () => {
+		const notOpenCategoryPanels = await page.$$(
+			'.editor-inserter__results .components-panel__body:not(.is-opened)'
+		);
+		for ( const categoryPanel of notOpenCategoryPanels ) {
+			await categoryPanel.click();
+		}
+	};
+
+	const getAllInserterItemTitles = async () => {
+		const inserterItemTitles = await page.evaluate( () => {
+			return Array.from(
+				document.querySelectorAll(
+					'.editor-inserter__results .editor-block-types-list__item-title'
+				)
+			).map(
+				( inserterItem ) => {
+					return inserterItem.innerText;
+				}
+			);
+		} );
+		return sortBy( uniq( inserterItemTitles ) );
+	};
+
+	beforeEach( async () => {
+		await createNewPost();
+	} );
+
+	it( 'all blocks should appear', async () => {
+		await page.click( '.editor-default-block-appender' );
+		await openGlobalInserter();
+		await openAllInserterCategories();
+		expect( await getAllInserterItemTitles() ).toMatchSnapshot();
+	} );
+
+	it( 'media & text block should restrict allowed blocks', async () => {
```

Comment:
This test is very specific to the Media & Text block. It's fine to start with but I would personally prefer to see this kind of tests inside `blocks/media-text.spec.js` file. It's faster to start with such verification but ideally, we use some custom block registered for testing purpose.